sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/orc/OrcPartitionReaderFactory.scala

```diff
@@ -163,13 +166,17 @@ case class OrcPartitionReaderFactory(
 
   private def createORCReader(filePath: Path, conf: Configuration): Reader = {
```

Comment:
How about letting the existing createORCReader method return a (Reader, OrcTail)? Or can we rebuild a usable OrcTail from Reader? For example, new OrcTail(reader.getFileTail, reader.getSerializedFileFooter)?